online-installer/install.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+#
+# Adopted from oh-my-zsh standalone installation script which is licensed under MIT License.
+#
+# This script should be run via curl:
+#   bash -c "$(curl -fsSL https://raw.githubusercontent.com/ioi-2017/tps/master/online-installer/install.sh)"
+# or via wget:
+#   bash -c "$(wget -qO- https://raw.githubusercontent.com/ioi-2017/tps/master/online-installer/install.sh)"
+# or via fetch:
+#   bash -c "$(fetch -o - https://raw.githubusercontent.com/ioi-2017/tps/master/online-installer/install.sh)"
+#
+# As an alternative, you can first download the install script and run it afterwards:
+#   wget https://raw.githubusercontent.com/ioi-2017/tps/master/online-installer/install.sh
+#   bash online-installer/install.sh
+
+set -euo pipefail
+
+# Make sure important variables exist if not already defined
+#
+# $USER is defined by login(1) which is not always executed (e.g. containers)
+# POSIX: https://pubs.opengroup.org/onlinepubs/009695299/utilities/id.html
+USER=${USER:-$(id -u -n)}
+# $HOME is defined at the time of login, but it could be unset. If it is unset,
+# a tilde by itself (~) will not be expanded to the current user's home directory.
+# POSIX: https://pubs.opengroup.org/onlinepubs/009696899/basedefs/xbd_chap08.html#tag_08_03
+HOME="${HOME:-$(getent passwd $USER 2>/dev/null | cut -d: -f6)}"
+# macOS does not have getent, but this works even if $HOME is unset
+HOME="${HOME:-$(eval echo ~$USER)}"
+
+# Directory of TPS code repository
+TPS="${TPS:-$HOME/.local/share/tps}"
+
+# Tps cli file path. It will be a symlink from the code repository and the directory should be added to the path.
+TPS_BIN="${TPS_BIN:-$HOME/.local/bin/tps}"
+REPO=${REPO:-ioi-2017/tps}
+REMOTE=${REMOTE:-https://github.com/${REPO}.git}
+BRANCH=${BRANCH:-master}
```

Comment:
I suggest inlining `REPO` into `REMOTE` as some users might want to use non-github repositories.

Additionally, the name of these environment variables seems to be too general and might be defined for other purposes which may cause problems here.
I recommend names such as `TPS_REMOTE_REPO` and `TPS_REMOTE_BRANCH`.

The name of variable `TPS` is also too general and ambiguous.
Something like `TPS_LOCAL_REPO` seems to be better.

Reply:
Makes sense.